sql/core/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -55,7 +56,7 @@ object SQLConf {
   val WAREHOUSE_PATH = SQLConfigBuilder("spark.sql.warehouse.dir")
     .doc("The default location for managed databases and tables.")
     .stringConf
-    .createWithDefault("file:${system:user.dir}/spark-warehouse")
+    .createWithDefault("${system:user.dir}/spark-warehouse")
```

Comment:
If the goal is to have a local directory here, always, you could add this to the config constant:

```
.transform(new File(_).toURI().toString())
```

Which should be Windows-compatible, right?